File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int ans = 0;
        int counter = 0;
        for(int i=0; i<nums.size(); i++)
        {
            if(nums[i]==1)
            {
                counter++;
                ans = max(ans,counter);
            }
            else{
                counter = 0;
            }
        }
        return ans;
    }
};

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int n = nums.length; 
       int maxStrike = Integer.MIN_VALUE; 
       int currStrike = 0;
    
       for(int i=0; i<n; i++){
            if(nums[i] == 1){
                currStrike++;
            }else{
                currStrike = 0;
            }
            maxStrike = Math.max(currStrike, maxStrike);
       } 

       return maxStrike;
    }
}